scripts/bqeth_subscription.py

```diff
@@ -0,0 +1,267 @@
+        address=GLOBAL_ALLOW_LIST_CONTRACT_ADDRESS, abi=GLOBAL_ALLOW_LIST_CONTRACT_ABI
+    )
+    # click.echo(f"ERC20 contract: {erc20_contract.address}")
+    click.echo(f"Subscription contract: {subscription_contract.address}")
+    click.echo(f"Coordinator contract: {coordinator_contract.address}")
+    click.echo(f"Global allow list contract: {global_allow_list_contract.address}")
+    # return erc20_contract, subscription_contract, coordinator_contract, global_allow_list_contract
+    return subscription_contract, subscription_contract, coordinator_contract, global_allow_list_contract
+
+
+def setup_context(func):
+    """Decorator to set up connections and contracts before executing the command function."""
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        w3, account = setup_connections()
+        (
+            erc20_contract,
+            subscription_contract,
+            coordinator_contract,
+            global_allow_list_contract,
+        ) = setup_contracts(w3)
+        context = Context(
+            w3,
+            account,
+            erc20_contract,
+            subscription_contract,
+            coordinator_contract,
+            global_allow_list_contract,
+        )
+        return func(context, *args, **kwargs)
+
+    return wrapper
+
+
+@click.group()
+def cli():
+    """BqETH Subscription CLI"""
+    pass
+
+
+@cli.command()
+@click.option("--encryptor-slots", default=2, help="Number of encryptor slots to pay for.")
+@setup_context
+def pay_subscription(context, encryptor_slots):
+    """Pay for a new subscription period and initial encryptor slots."""
+    base_fees = context.subscription_contract.functions.baseFees(0).call()
+    encryptor_fees = context.subscription_contract.functions.encryptorFees(
+        MAX_NODES, context.subscription_contract.functions.subscriptionPeriodDuration().call()
+    ).call()
+
+    click.echo(
+        f"Approving transfer of {base_fees + encryptor_fees} ERC20 token for subscription contract."
+    )
+    tx_hash = context.erc20_contract.functions.approve(
+        SUBSCRIPTION_CONTRACT_ADDRESS, base_fees + encryptor_fees
+    ).transact({"from": context.account.address})
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+    click.echo(f"Paying for a new subscription period with {encryptor_slots} encryptor slots.")
+    tx_hash = context.subscription_contract.functions.payForSubscription(encryptor_slots).transact(
+        {"from": context.account.address}
+    )
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+
+@cli.command()
+@click.option("--extra-slots", default=1, help="Number of additional encryptor slots to pay for.")
+@setup_context
+def pay_slots(context, extra_slots):
+    """Pay for additional encryptor slots."""
+    click.echo(f"Paying for {extra_slots} new encryptor slots.")
+    tx_hash = context.subscription_contract.functions.payForEncryptorSlots(extra_slots).transact(
+        {"from": context.account.address}
+    )
```

Comment:
1. Does this function also imply the transfer of erc20 fees to the subscription contract?  
2. If so - we may need to include erc20 approval here also?  
3. Perhaps this is related to how you handle `MAX_NODES` above as a means of "pre-approving" an ERC20 amount?